apps/emqx_resource/src/emqx_resource_buffer_worker.erl

```diff
@@ -64,8 +64,10 @@
 
 -define(COLLECT_REQ_LIMIT, 1000).
 -define(SEND_REQ(FROM, REQUEST), {'$send_req', FROM, REQUEST}).
--define(QUERY(FROM, REQUEST, SENT, EXPIRE_AT), {query, FROM, REQUEST, SENT, EXPIRE_AT}).
--define(SIMPLE_QUERY(FROM, REQUEST), ?QUERY(FROM, REQUEST, false, infinity)).
+-define(QUERY(FROM, REQUEST, SENT, EXPIRE_AT, RULE_ID),
+    {query, FROM, REQUEST, SENT, EXPIRE_AT, RULE_ID}
```

Comment:
It should be a more generic trace context instead of RULE_ID.
```suggestion
    {query, FROM, REQUEST, SENT, EXPIRE_AT, TRACE_CONETXT}
```
`TRACE_CONTEXT` is either `undefined` (when no matching trace), or a map with the matching IDs (e.g. `clientid` and `ruleid`).

Reply:
There is never more than one rule_id per `?QUERY` but I think its worth making it more easy to extend anyway. For example, we need to forward information about if the action should be stopped after the template is rendered and I think this info can be passed in the `TRACE_CONTEXT` as well.

Reply:
> i.e. one should be able to trace rules with clientid fitler.

Ok, I see, so I will add client_id to the trace message's meta data.

Reply:
@zmstone  As you suggested, I have made this field into TRACE_CONTEXT and added clientid so that that rules can be traced using the clientid filter.